peer/brontide.go

```diff
@@ -2752,12 +2766,28 @@ func (p *Brontide) RemoteFeatures() *lnwire.FeatureVector {
 	return p.remoteFeatures
 }
 
-// sendInitMsg sends the Init message to the remote peer. This message contains our
-// currently supported local and global features.
-func (p *Brontide) sendInitMsg() error {
+// sendInitMsg sends the Init message to the remote peer. This message contains
+// our currently supported local and global features.
+func (p *Brontide) sendInitMsg(legacyChan bool) error {
+	features := p.cfg.Features
+
+	// If we have a legacy channel open with a peer, we downgrade static
+	// remote required to optional in case the peer does not understand the
+	// required feature bit. If we do not do this, the peer will reject our
+	// connection because it does not understand a required feature bit, and
+	// our channel will be unusable.
+	if legacyChan && features.RequiresFeature(lnwire.StaticRemoteKeyRequired) {
+		peerLog.Infof("legacy channel open with peer: %x, "+
+			"downgrading static remote required feature bit to "+
+			"optional", p.PubKey())
+
+		features.Unset(lnwire.StaticRemoteKeyRequired)
```

Comment:
I think I checked this, but just to make sure: Is `features` a local copy at this point, or do we risk unsetting the bit for our features for future handshakes? Could be worth calling `features.Clone()` here just to be sure :)

Reply:
It's a pointer, we need to clone.

Reply:
done!

Reply:
FWIW `cfg.Features` is already a local copy, we clone a fresh feature vector for each connection from the static feature definitions 
https://github.com/lightningnetwork/lnd/blob/304a25691f14db448a55a6db55953c3864b7d80b/server.go#L3054
https://github.com/lightningnetwork/lnd/blob/304a25691f14db448a55a6db55953c3864b7d80b/feature/manager.go#L121
https://github.com/lightningnetwork/lnd/blob/304a25691f14db448a55a6db55953c3864b7d80b/feature/manager.go#L113